src/discord2telegram/setup.ts

```diff
@@ -1,402 +1,455 @@
+				.then(sleepOneMinute)
+				.then((info: any) => Promise.all([info.delete(), message.delete()]))
+				.catch(ignoreAlreadyDeletedError);
+
+			// Don't process the message any further
+			return;
+		}
+
+		// Get info about the sender
+		const senderName = R.compose<any, any>(
+			// Make it HTML safe
+			escapeHTMLSpecialChars,
+			// Add a colon if wanted
+			//@ts-ignore
+			R.when(R.always(settings.telegram.colonAfterSenderName), senderName => senderName + ":"),
+			// Figure out what name to use
+			R.ifElse(
+				message => useNickname && !R.isNil(message.member),
+				R.path(["member", "displayName"]),
+				R.path(["author", "username"])
+			)
+		)(message) as string;
+
+		// Check if the message is from the correct chat
+		const bridges = bridgeMap.fromDiscordChannelId(Number(message.channel.id));
+		if (!R.isEmpty(bridges)) {
+			bridges.forEach(async bridge => {
+				// Ignore it if this is a telegram-to-discord bridge
+				if (bridge.direction === Bridge.DIRECTION_TELEGRAM_TO_DISCORD) {
+					return;
+				}
+
+				// This is now the latest message for this bridge
+				latestDiscordMessageIds.setLatest(message.id, bridge);
+
+				// Check if the message is a reply and get the id of that message on Telegram
+				let replyId = 0;
+				const messageReference = message?.reference;
+
+				if (typeof messageReference !== "undefined") {
+					const referenceId = messageReference?.messageId;
+					if (typeof referenceId !== "undefined") {
+						console.log("==== discord2telegram/setup.ts reply ====");
+						console.log("referenceId: " + referenceId);
+						console.log("bridge.name: "+ bridge.name);
+						[replyId] = messageMap.getCorrespondingReverse(MessageMap.TELEGRAM_TO_DISCORD, bridge, referenceId as string);
+						console.log("t2d replyId: " + replyId);
+						if (isNaN(replyId)) {
+							[replyId] = messageMap.getCorresponding(MessageMap.DISCORD_TO_TELEGRAM, bridge, referenceId as string);
+							console.log("d2t replyId: " + replyId);
+						}
+					}
+				}
+
+				// Check for attachments and pass them on
+				message.attachments.forEach(async ({ url }) => {
+					try {
+						const textToSend = bridge.discord.sendUsernames
+							? `<b>${senderName}</b>\n<a href="${url}">${url}</a>`
+							: `<a href="${url}">${url}</a>`;
+						if (replyId === 0 || replyId === undefined) {
+							const tgMessage = await tgBot.telegram.sendMessage(bridge.telegram.chatId, textToSend, {
+								parse_mode: "HTML"
+							});
+							messageMap.insert(
+								MessageMap.DISCORD_TO_TELEGRAM,
+								bridge,
+								message.id,
+								tgMessage.message_id.toString()
+							);
+						} else {
+							const tgMessage = await tgBot.telegram.sendMessage(bridge.telegram.chatId, textToSend, {
+								reply_to_message_id: replyId,
+								parse_mode: "HTML"
+							});
+							messageMap.insert(
+								MessageMap.DISCORD_TO_TELEGRAM,
+								bridge,
+								message.id,
+								tgMessage.message_id.toString()
+							);
+						}

```

Comment:
This is equal in both the if and else, so it can be moved out.
```suggestion
						} else {
							const tgMessage = await tgBot.telegram.sendMessage(bridge.telegram.chatId, textToSend, {
								reply_to_message_id: replyId,
								parse_mode: "HTML"
							});
						}
						messageMap.insert(
							MessageMap.DISCORD_TO_TELEGRAM,
							bridge,
							message.id,
							tgMessage.message_id.toString()
						);
```

Reply:
Actually can't move it out because at the moment, the tgMessage declaration happens inside the if/else, moving it out would make it throw an error

Reply:
still in discussion?

Reply:
Nope, let me close it